lib/facility.rb

```diff
@@ -1,14 +1,109 @@
+require './lib/vehicle'
+require './lib/registrant'
+require './lib/dmv_data_service'
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :registered_vehicles, :collected_fees, :operating_hours, :holiday_closures
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(info)
+    @name = info[:name]
+    @address = info[:address]
+    @phone = info[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
+    @operating_hours = info[:operating_hours] || {}
+    @holiday_closures = info[:holiday_closures] || []
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if @services.include?('Vehicle Registration')
+      vehicle.registration_date = Date.today
+      vehicle.plate_type = set_plate_type(vehicle)
```

Comment:
Since you're assigning the `vehicle.plate_type` on lines 40, 43, or 46, this line should probably just call the method:
```
...
vehicle.registration_date = Date.today
set_plate_type(vehicle)
@registered_vehicles << vehicle
...
```